### PR TITLE
ci: add cross-version interop

### PR DESCRIPTION
**Pull request by laurentsenta**
# Description

Update workflows on push & PR:

* `run-ping-interop-cross-version`: runs a Testground interoperability test between multiple versions of go-libp2p, including master, and the current branch (during a pull request)
* `run-ping-interop-cross-implementation`: runs a Testground interoperability test between go-libp2p and rust-libp2p, including master, and the current branch (during a pull request)

We rely on the https://github.com/libp2p/test-plans/ repository to retrieve and run the tests.

Demonstration:

* passing PR: https://github.com/laurentsenta/go-libp2p/pull/6
* breaking PR: https://github.com/laurentsenta/go-libp2p/pull/5

### CODE REVIEW

**Comment by laurentsenta** [comment]
@marten-seemann interop test - take two :)
Could we re-enable the tests?

File: .github/workflows/interop-test.yml
Comment (marten-seemann) [inline .github/workflows/interop-test.yml:14]: Why are you deleting this comment?

File: .github/workflows/interop-test.yml
Comment (marten-seemann) [inline .github/workflows/interop-test.yml:17]: Is this the test that takes 70 minutes that you talked about yesterday?

File: .github/workflows/interop-test.yml
Comment (laurentsenta) [reply .github/workflows/interop-test.yml:14]: I moved it to line 5

Comment (marten-seemann) [reply .github/workflows/interop-test.yml:14]: Sorry! How could I miss that?! 🤦🏻‍♂️

File: .github/workflows/interop-test.yml
Comment (laurentsenta) [reply .github/workflows/interop-test.yml:17]: No no, there is no test that takes 70 minutes,

This test takes ~17 minutes. It test go-libp2p master + rust-libp2p master + the current branch.
There is a larger test that takes around ~27 minutes that tests all known versions together + the current branch but we won't enable it yet.

See the results here:
https://github.com/libp2p/test-plans/pull/32#issuecomment-1225994414

Comment (laurentsenta) [reply .github/workflows/interop-test.yml:17]: I shared 2 examples pull requests in the description if you want to see a recent run using this code:

passing: https://github.com/laurentsenta/go-libp2p/actions/runs/2955179619
breaking: https://github.com/laurentsenta/go-libp2p/actions/runs/2955183270

**Review by galargh — APPROVED** [review]
Looks good to me but please wait for @marten-seemann 's approval 🤝

**Pull request merged**
